datajunction-query/djqs/engine.py

```diff
@@ -66,22 +71,61 @@ def get_columns_from_description(
     return columns
 
 
+def create_trino_engine(engine: Engine, headers: Optional[Dict[str, str]]) -> "Engine":
+    """
+    Create a SQLAlchemy engine for Trino.
+    """
+    # Parsing the URI
+    parsed_uri = urlparse(engine.uri)
+    scheme = parsed_uri.scheme
+    host = parsed_uri.hostname
+    port = parsed_uri.port
+    path = parsed_uri.path.lstrip("/")
+    catalog, schema = path.split("/", 1) if "/" in path else (path, None)
+
+    global QUERY_USER, QUERY_PASSWORD, HTTP_SCHEME
+
+    # Override user and password from headers if available
+    if headers and "QUERY_USER" in headers and "QUERY_PASSWORD" in headers:
+        QUERY_USER = headers.get("QUERY_USER")
+        QUERY_PASSWORD = headers.get("QUERY_PASSWORD")
+
+    if HTTP_SCHEME == "https":
+        engine_str = f"{scheme}://{QUERY_USER}:{QUERY_PASSWORD}@{host}:{port}/{catalog}/{schema}?protocol={HTTP_SCHEME}"
+        return create_engine(engine_str)
+    else:
+        engine_str = engine.uri
+        return create_engine(engine_str, connect_args=engine.extra_params)
+
+
 def run_query(
     session: Session,
     query: Query,
+    headers: Optional[Dict[str, str]] = None,
 ) -> List[Tuple[str, List[ColumnMetadata], Stream]]:
     """
     Run a query and return its results.
 
     For each statement we return a tuple with the statement SQL, a description of the
     columns (name and type) and a stream of rows (tuples).
     """
+
     _logger.info("Running query on catalog %s", query.catalog_name)
+
     engine = session.exec(
         select(Engine)
         .where(Engine.name == query.engine_name)
         .where(Engine.version == query.engine_version),
     ).one()
+
+    parsed_uri = urlparse(engine.uri)
+    scheme = parsed_uri.scheme
+
+    global QUERY_USER, QUERY_PASSWORD, HTTP_SCHEME
+    QUERY_USER = engine.extra_params["user"]
+    QUERY_PASSWORD = engine.extra_params["password"]
+    HTTP_SCHEME = "https"
```

Comment:
Instead of hard-coding this here, can you include that in the request headers and then parse them out here? `http_scheme = headers.get("TRINO_HTTP_SCHEME")`